packages/jest-cli/src/__tests__/TestRunner-test.js

```diff
@@ -32,9 +32,16 @@ jest.mock('../reporters/DefaultReporter');
 test('.addReporter() .removeReporter()', () => {
   const runner = new TestRunner({}, {}, {});
   const reporter = new SummaryReporter();
+
+  expect(runner._dispatcher._reporters).toHaveLength(2);
   runner.addReporter(reporter);
+
   expect(runner._dispatcher._reporters).toContain(reporter);
+  expect(runner._dispatcher._reporters).toHaveLength(3);  
+
   runner.removeReporter(SummaryReporter);
+
+  expect(runner._dispatcher._reporters).toHaveLength(2);
```

Comment:
since removing reporters works using `instanceOf`, both instances of `SummareReporters` will be removed, and the length here will be `1` (test is currently failing on the CI)